static/js/typeahead_helper.js

```diff
@@ -207,17 +207,27 @@ exports.compare_by_sub_count = function (stream_a, stream_b) {
     return stream_a.subscribers.num_items() < stream_b.subscribers.num_items();
 };
 
+// Find which stream comes first in the sidebar listing order
+exports.compare_by_sidebar = function (stream_a, stream_b) {
+    var a_score = stream_data.is_active(stream_a.name) + stream_a.pin_to_top * 2;
+    var b_score = stream_data.is_active(stream_b.name) + stream_b.pin_to_top * 2;
```

Comment:
By score function, would you mean something like:
`function score (stream) {
    if (stream.pin_to_top) {
        return 2;
    }
    if (stream_data.is_active(stream.name)) {
        return 1;
    } 
    return 0;
}`
It's quite a bit more bulky, but would you prefer something like that to a solution implicity treating bools as ints?